src/napari_micromanager/_gui_objects/_mda_widget.py

```diff
@@ -22,36 +30,35 @@ def __init__(
     ) -> None:
         super().__init__(include_run_button=True, parent=parent, mmcore=mmcore)
 
+        # add save widget
+        v_layout = cast(QVBoxLayout, self._central_widget.layout())
         self._save_groupbox = SaveWidget()
         self._save_groupbox.setSizePolicy(
             QSizePolicy.Policy.Minimum, QSizePolicy.Policy.Fixed
         )
         self._save_groupbox.setChecked(False)
-
-        v_layout = cast(QVBoxLayout, self.layout())
+        self._save_groupbox.toggled.connect(self._on_save_toggled)
+        self._save_groupbox._directory.textChanged.connect(self._on_save_toggled)
+        self._save_groupbox._fname.textChanged.connect(self._on_save_toggled)
         v_layout.insertWidget(0, self._save_groupbox)
 
-        self.channel_groupbox.setMinimumHeight(230)
+        # add split channel checkbox
+        self.channel_widget.setMinimumHeight(230)
         self.checkBox_split_channels = QCheckBox(text="Split Channels")
         self.checkBox_split_channels.toggled.connect(self._toggle_split_channel)
-        g_layout = cast(QGridLayout, self.channel_groupbox.layout())
+        g_layout = cast(QGridLayout, self.channel_widget.layout())
         g_layout.addWidget(self.checkBox_split_channels, 1, 0)
-
-        # TODO: stage_pos_groupbox should have a valueChanged signal
-        # and that should be connected to _toggle_checkbox_save_pos
-        self._save_groupbox.toggled.connect(self._toggle_checkbox_save_pos)
-        self.position_groupbox.valueChanged.connect(self._toggle_checkbox_save_pos)
-        self.channel_groupbox.valueChanged.connect(self._toggle_split_channel)
+        self.channel_widget.valueChanged.connect(self._toggle_split_channel)
 
     def _toggle_split_channel(self) -> None:
-        if not self.channel_groupbox.value():
-            self.checkBox_split_channels.setChecked(False)
-
-    def _toggle_checkbox_save_pos(self) -> None:
         if (
-            self.position_groupbox.isChecked()
-            and len(self.position_groupbox.value()) > 0
+            not self.channel_widget.value()
+            or self.channel_widget._table.rowCount() == 1
```

Comment:
this is a usage of a private widget in pymmcore-widgets.  If there is literally no public way to access this and you need it here, then you should make it public in pymmcore-widgets